R/get_dataset_versions.R

```diff
@@ -1,20 +1,53 @@
 #' Get data set versions
 #'
 #' @inheritParams api_url
+#' @param detail Level of detail to return. Given as a character string, it should be one of:
+#' "light" (default) or "full".
+#' Using "light" gives the following:
+#' \itemize{
+#'   \item version - version number
+#'   \item type - version type (major or minor)
+#'   \item total_rows - total number of rows in the listed version
+#'   \item date_released - release date of the version
+#'   \item version_title - name of the version release
+#'   \item time_period_start - first available time period in the data
+#'   \item time_period_start - latest available time period in the data
+#' }
+#'
+#' Using "full" adds the following in addition to the above:
+#' \itemize{
+#'   \item version_id - unique ID code for the data set version
+#'   \item status - publication status of the data set version
+#'   \item geographic_levels - geographic levels provided
+#'   \item filters - names of filters in the data set version
+#'   \item indicators - names of indicators in the data set version
+#'   \item notes - any additional information
+#'   }
+#'
+#' @return A data frame with X columns:
 #'
 #' @return Data frame listing all available versions of the given data set
 #' @export
 #'
 #' @examples
-#' get_dataset_versions(dataset_id = example_id(group="attendance"))
+#' get_dataset_versions(dataset_id = example_id(group = "attendance"))
 get_dataset_versions <- function(
     dataset_id,
+    detail = "light",
     ees_environment = NULL,
     api_version = NULL,
     page_size = 40,
     page = NULL,
     verbose = FALSE) {
+  # Do some basic validation
   validate_page_size(page_size)
+
+  detail <- tolower(detail)
+  if (!(detail %in% c("light", "full"))) {
+    stop("The detail parameter should be either \"light\" or \"full\". Value passed: ", detail)
```

Comment:
Nice! We should do the printing of the offending value more often in our error messages